mrbgems/mruby-compiler/core/parse.y

```diff
@@ -4381,12 +4381,7 @@ parse_string(parser_state *p)
           }
         }
         if ((len-1 == hinf->term_len) && (strncmp(s, hinf->term, len-1) == 0)) {
-          if (c < 0) {
-            p->parsing_heredoc = NULL;
-          }
-          else {
-            return tHEREDOC_END;
-          }
+          return tHEREDOC_END;
```

Comment:
Should it be the following, just in case parsing continues after `EOF` using `partial_hook`?
```C
if (c<0) {
  p->parsing_heredoc = NUL;
}
return tHEREDOC_END;
```

Reply:
Thank you for your comment.

I think `p->parsing_heredoc` will become `NULL` via `heredoc_end()`. At least the above code will crash at:

https://github.com/mruby/mruby/blob/db43738397c2236adbdd9750db971106d5335ab6/mrbgems/mruby-compiler/core/parse.y#L1276

Reply:
I see. Thank you for the input.